controllers/capability/routing/reconciler.go

```diff
@@ -94,11 +95,16 @@ func (r *Reconciler) Reconcile() (update bool, err error) {
 }
 
 func (r *Reconciler) createServiceIfNotExists() (bool, error) {
-	service := createService(r.Instance, module)
-	err := r.Get(context.TODO(), client.ObjectKey{Name: service.Name, Namespace: service.Namespace}, &service)
+	service := createService(r.Instance, Module)
+
+	if err := controllerutil.SetControllerReference(r.Instance, service, r.Scheme()); err != nil {
```

Comment:
Technically you only need to set this reference when creating, i.e., you didn't find it.

As it's now, you're doing this operation even if you create it or not.

tldr: put it inside the if below.